File: 2022/day1/star2.cpp
#include <iostream>
#include <fstream>
#include <string>
#include <vector>
#include <algorithm>st
using namespace std;

int main() {
    string line;
    ifstream input("input.txt");
    int cur = 0;
    vector<int> v;
    while(getline(input, line)) {
        if(line.empty()) {
            v.push_back(cur);
            cur = 0;
        }
        else {
            cur+=stoi(line);
        }
    }
    v.push_back(cur);
    cur =0;
    sort(v.begin(), v.end(), greater<int>());
    
    for(int i =0; i<3; i++) {
        cur+=v[i];
    }
    cout << cur << endl;
}

File: 2025/day1/star2.py
cur = 50
cnt = 0
prev = None
with open ("2025/day1/input.txt") as f:
    for line in f:
        sign = 1 if line[0]=="R" else -1
        d = int(line[1:])
        remaining_steps = (100 - cur) if sign == 1 else cur
        if remaining_steps == 0:
            remaining_steps = 100
        if remaining_steps <=d:
            cnt+= 1+(d-remaining_steps)//100
        cur+=sign*d
        cur%=100


print(cnt)
